src/class.js

```diff
@@ -0,0 +1,45 @@
+export default class Tasks {
+  constructor() {
+    this.list = localStorage.getItem('tasks')
+      ? JSON.parse(localStorage.getItem('tasks'))
+      : [];
+  }
+
+  add(task) {
+    task.index = this.list.length + 1;
+    task.completed = false;
+    this.list.push(task);
+    localStorage.setItem('tasks', JSON.stringify(this.list));
+  }
+
+  remove(index) {
+    this.list = this.list.filter((task) => task.index !== index);
+    this.list = this.list.map((t) => {
+      if (t.index > index) {
+        t.index -= 1;
+      }
+      return t;
+    });
+    localStorage.setItem('tasks', JSON.stringify(this.list));
+  }
+
+  edit(task) {
+    this.list[task.index - 1] = task;
+    localStorage.setItem('tasks', JSON.stringify(this.list));
+  }
+
+  sort(oldIndex, newIndex) {
+    this.list[oldIndex - 1].index = newIndex;
+    localStorage.setItem('tasks', JSON.stringify(this.list));
+  }
+
+  clearCompleted() {
+    this.list = this.list.sort((a, b) => a.index - b.index);
+    this.list = this.list.filter((t) => !t.completed);
```

Comment:
- [ ] _[OPTIONAL]_ Filter the array before sorting if u wanna reset the indexes. this should take one less iteration per deleted task.
- [ ] The `clear all completed` button isn't working (no console errors and the localstorage isn't updated).

Reply:
the `clear all completed` is working for me. make sure you mark some tasks as complete first.

https://user-images.githubusercontent.com/42470159/149208428-a6c70e91-3664-4881-8ec5-86821c81ca85.mp4